flocker/node/agents/test/test_blockdevice.py

```diff
@@ -3109,6 +3109,42 @@ class IBlockDeviceAPITestsMixin(object):
     """
     this_node = None
 
+    def repeat_retries(self):
+        """
+        @return: An iterable of delay intervals, measured in seconds, used
+             to retry in ``repeat_until_consistent``. By default only one
+             try is allowed; subclasses can override to change this
+             policy.
+        """
+        return [0]
+
+    def repeat_until_consistent(self, f, *args, **kwargs):
+        """
+        Repeatedly call a function with given arguments until AssertionErrors
+        stop or configured number of repetitions are reached.
+
+        Some backends are eventually consistent, which means results of
+        listing volumes may not reflect actions immediately. So for
+        read-only operations that rely on listing we want to be able to
+        retry.
+
+        Retry policy can be changed by overriding the ``repeat_retries``
+        method.
+
+        @param f: Function to call.
+        @param args: Arguments for ``f``.
+        @param kwargs: Keyword arguments for ``f``.
+        """
+        for step in self.repeat_retries():
+            try:
+                f(*args, **kwargs)
+            except AssertionError as e:
+                pass
+            else:
+                return
+            time.sleep(step)
+        raise e
+
```

Comment:
Once I implement all the necessary boilerplate it's no shorter, sadly, so not going to bother merging rewritten version.